papis/commands/doctor.py

```diff
@@ -0,0 +1,374 @@
+                             suggestion_cmd=("papis edit --doc-folder {}"
+                                             .format(folder)),
+                             fix_action=lambda: None,
+                             payload=doc["type"],
+                             doc=doc))
+    return results
+
+
+HTML_CODE_REGEX = re.compile(r"&[a-z_A-Z0-9]+;")
+HTML_CODES_CHECK_NAME = "html-codes"
+
+
+def html_codes_check(doc: papis.document.Document) -> List[Error]:
+    """
+    Checks that the keys in "doctor-html-code-keys" do not contain
+    any html codes like &amp; etc.
+    """
+    results = []
+    folder = doc.get_main_folder()
+
+    def _fix(key: str) -> Callable[[], None]:
+
+        def __fix() -> None:
+            db = papis.database.get()
+            val = html.unescape(doc[key])
+            logger.info("Setting '%s' to '%s'", key, val)
+            doc[key] = val
+            doc.save()
+            db.update(doc)
+
+        return __fix
+
+    for key in papis.config.getlist("doctor-html-codes-keys"):
+        if doc[key]:
+            m = HTML_CODE_REGEX.findall(str(doc[key]))
+            if m:
+                results.append(Error(name=HTML_CODES_CHECK_NAME,
+                                     path=folder or "",
+                                     msg=("Field '{}' contains html codes '{}'"
+                                          .format(key, m)),
+                                     suggestion_cmd=("papis edit "
+                                                     "--doc-folder {}"
+                                                     .format(folder)),
+                                     fix_action=_fix(key),
+                                     payload=key,
+                                     doc=doc))
+    return results
+
+
+REGISTERED_CHECKS = {}  # type: Dict[str, Check]
+register_check(FILES_CHECK_NAME, files_check)
+register_check(KEYS_EXIST_CHECK_NAME, keys_check)
+register_check(DUPLICATED_KEYS_NAME, duplicated_keys_check)
+register_check(BIBTEX_TYPE_CHECK_NAME, bibtex_type_check)
+register_check(REFS_CHECK_NAME, refs_check)
+register_check(HTML_CODES_CHECK_NAME, html_codes_check)
+
+
+def run(doc: papis.document.Document, checks: List[str]) -> List[Error]:
+    """
+    Runner for doctor. It runs all the checks given by the check
+    argument, and it gets the check from the global REGISTERED_CHECKS
+    dictionary.
+    """
+    results = []  # type: List[Error]
+    for check in checks:
+        results.extend(REGISTERED_CHECKS[check].operate(doc))
+    return results
+
+
+@click.command("doctor")
+@click.help_option("--help", "-h")
+@papis.cli.query_option()
+@papis.cli.sort_option()
+@click.option("-t", "--checks", "_checks",
+              default=lambda: papis.config.getlist("doctor-default-checks"),
+              multiple=True,
+              help=("Checks to run on every document, possible values: {}"
+                    .format(", ".join(registered_checks_names()))))
```

Comment:
Do a `click.Choice` for the check names?

Reply:
Yes that is a good idea